src/ipc/ipc-zephyr.c

```diff
@@ -133,13 +133,24 @@ enum task_state ipc_platform_do_cmd(struct ipc *ipc)
 	/* perform command */
 	ipc_cmd(hdr);
 
-#if CONFIG_SOC_SERIES_INTEL_ADSP_CAVS
-	/* are we about to enter D3 ? */
-	if (ipc->pm_prepare_D3) {
-		/* no return - memory will be powered off and IPC sent */
+	if (ipc->task_mask & IPC_TASK_POWERDOWN) {
+#if defined(CONFIG_PM)
```

Comment:
CONFIG_PM is a bit too generic as it's enabled for all platforms even if core power state is still managed on SOF side (like it is for TGL).

How about use #ifdef CONFIG_PM_POLICY_CUSTOM ... that's only set if we use the custom SOF policy?